From efc5353500f1a3e09d831beeb1c0ce32e4f24aee Mon Sep 17 00:00:00 2001
From: Matthew Stewart <Matthew.Stewart2@amd.com>
Date: Thu, 16 Jul 2026 15:07:35 -0400
Subject: [PATCH 15/16] drm/amdgpu: Switch order of GC and Display IP blocks

For DCN42B, GC hardware init should occur before display HW init.
Otherwise, the GB_ADDR_CONFIG register accessed in
amdgpu_dm_plane_add_gfx11_modifiers will have a value of 0, causing
plane modifiers to not be set correctly.

Signed-off-by: Matthew Stewart <Matthew.Stewart2@amd.com>
Reviewed-by: Aurabindo Pillai <aurabindo.pillai@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_discovery.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_discovery.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_discovery.c
index 2860f12..fa5ebe0 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_discovery.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_discovery.c
@@ -3427,11 +3427,11 @@ int amdgpu_discovery_set_ip_blocks(struct amdgpu_device *adev)
 			return r;
 	}
 
-	r = amdgpu_discovery_set_display_ip_blocks(adev);
+	r = amdgpu_discovery_set_gc_ip_blocks(adev);
 	if (r)
 		return r;
 
-	r = amdgpu_discovery_set_gc_ip_blocks(adev);
+	r = amdgpu_discovery_set_display_ip_blocks(adev);
 	if (r)
 		return r;
 
-- 
2.55.0

